### PR TITLE
datastore cleanup

### PR DESCRIPTION
**Pull request by djosborne**
- Replaced references to "docker" in datastore by abstracting with orchestrator
- Modified other functions to use the new and optimized get_endpoints
- Removed obsolete shownodes and get_host functionality
See individual commits for easier view

### CODE REVIEW

File: calico_containers/pycalico/datastore.py [hunk 1]
```diff
@@ -1058,7 +1009,7 @@ def get_endpoint(self, hostname=None, orchestrator_id=None,
             return eps.pop()
 
     @handle_errors
-    def set_endpoint(self, hostname, container_id, endpoint):
+    def set_endpoint(self, hostname, orchestrator_id, container_id, endpoint):
```
Comment (robbrockbank) [inline calico_containers/pycalico/datastore.py]: Does set_endpoint actually need the various IDs ... I assume they are actually encapsulated in the endpoint object iteself and so all but the endpoint are superflous?

Comment (djosborne) [inline calico_containers/pycalico/datastore.py]: Makes sense to me.

**Comment by robbrockbank** [comment]
One comment about set_endpoint(), otherwise LGTM.

File: calico_containers/pycalico/datastore.py [hunk 2]
```diff
@@ -1058,16 +1020,23 @@ def get_endpoint(self, hostname=None, orchestrator_id=None,
             return eps.pop()
 
     @handle_errors
-    def set_endpoint(self, hostname, container_id, endpoint):
+    def set_endpoint(self, endpoint):
```
Comment (djosborne) [inline calico_containers/pycalico/datastore.py]: @robbrockbank these assertions a good  idea?

Comment (robbrockbank) [inline calico_containers/pycalico/datastore.py]: thye might be better in the constructor of the Endpoint.

**Comment by djosborne** [comment]
offline talk with Rob, going to make changes to set_endpoint in seperate PR due to complexity of change

**Pull request merged**